### PR TITLE
chore: add helper to register functions as actors

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change is fairly useful for creating dummy actors (for instance, a trial logger) while avoiding a lot of boiler plate.

## Test Plan

Zero impact change, but it will be tested by being used in tests.

## Commentary (optional)
Opening this separately because I don't want to just open some mega pr for all the tests I'm trying to write.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/pkg/actor/actor.go [hunk 1]
```diff
@@ -30,3 +30,10 @@ type Actor interface {
 	// a request to stop is received or the parent shuts the actor down.
 	Receive(context *Context) error
 }
+
+// ActorFunc is a function that encapsulates behavior. It is a stateless actor, useful for a mocking.
+type ActorFunc func(context *Context) error
```
Comment (stoksc) [inline master/pkg/actor/actor.go:35]: Allows something like
```
system.ActorOf("trialLogger", actor.ActorFunc(func(context *Context) error { ... }))
```

**Review by dzhu — APPROVED** [review]
Looks helpful, thanks!

File: master/pkg/actor/actor.go [hunk 2]
```diff
@@ -30,3 +30,10 @@ type Actor interface {
 	// a request to stop is received or the parent shuts the actor down.
 	Receive(context *Context) error
 }
+
+// ActorFunc is a function that encapsulates behavior. It is a stateless actor, useful for a mocking.
```
Comment (dzhu) [inline master/pkg/actor/actor.go:34]:
```suggestion
// ActorFunc is a function that encapsulates behavior. It is a stateless actor, useful for mocking.
```

Comment (stoksc) [reply master/pkg/actor/actor.go]: 🤦 somehow I didn't see this. I can fix this typo in my next PR.

**Pull request merged**